src/main/java/org/jenkinsci/plugins/ghprb/extensions/status/GhprbSimpleStatus.java

```diff
@@ -165,6 +174,11 @@ public void onBuildStart(AbstractBuild<?, ?> build,
         } else {
             sb.append(Ghprb.replaceMacros(build, listener, startedStatus));
         }
+
+        if(showMatrixStatus && !(build.getProject() instanceof MatrixProject)){
```

Comment:
Sorry about the confusing variable name. I'll rename it to something more appropriate. But as of now, ```showMatrixStatus = true``` when the checkbox is clicked, and ```false``` otherwise. So in a freestyle project with the box unchecked,  the condition for the if statement is false.